inference.py

```diff
@@ -0,0 +1,149 @@
+import argparse
+import torch
+import pandas as pd
+
+from fairseq import options, tasks, checkpoint_utils
+from fairseq.data import encoders
+from tqdm import tqdm
+
+import data
+
+
+def main(script_args, model_args):
+    split = script_args.split
+    predictions = predict(image_id_path='output/{}-ids.txt'.format(split),
+                          grid_features_path='output/{}-features-grid'.format(split),
+                          obj_features_path='output/{}-features-obj'.format(split),
+                          obj_features_meta_path='output/{}-features-obj/metadata.csv'.format(split),
+                          model_args=model_args)
```

Comment:
`output` directory shouldn't be hardcoded but rather 

```python
    predictions = predict(image_id_path=f'{model_args.captions_dir}/{split}-ids.txt',
                          grid_features_path=f'{model_args.features_dir}/{split}-features-grid',
                          obj_features_path=f'{model_args.features_dir}/{split}-features-obj',
                          obj_features_meta_path=f'{model_args.features_dir}/{split}-features-obj/metadata.csv',
                          model_args=model_args)
```

Also, let's do string interpolation consistently across the project e.g. use `f'... {foo}'` instead of `... {}`.format(foo) as already done in other modules. Or do you have specific reasons to prefer `.format()`?